File: Basics/OOPS/CPP/UPI.cpp
#include<iostream>
#include "PaymentMethod.cpp"
using namespace std;

class UPI :public PaymentMethod{
    private:
      string upiID;
    public: 
    UPI(const string& upiId){
      this->upiID = upiId;
    }
    void processPayment(double amount) override{
        cout << "Processing UPI payment of $" << amount << " for UPI ID: " << upiID << endl;
    }
};

File: Basics/OOPS/JAVA/UPI.java
public class UPI implements PaymentMethod {
    public String UPIId;
    UPI(String UPIId) {
        this.UPIId = UPIId;
    }

    @Override
    public void pay(double amount) {
        System.out.println("Paid " + amount + " using UPI: " + UPIId);  
    }
}
